src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/DataSetsControllerQueryTests.cs

```diff
@@ -0,0 +1,2093 @@
+
+            validationProblem.AssertHasSortMaxFieldLengthError(
+                expectedPath: "sorts[6]",
+                field: new string('a', 41)
+            );
+            validationProblem.AssertHasSortMaxFieldLengthError(
+                expectedPath: "sorts[7]",
+                field: new string('b', 41)
+            );
+        }
+
+        [Fact]
+        public async Task Sorts_FieldsNotFound_Returns400()
+        {
+            var dataSetVersion = await SetupDefaultDataSetVersion();
+
+            string[] notFoundSorts =
+            [
+                "invalid1|Asc",
+                "invalid2|Desc",
+            ];
+
+            var response = await QueryDataSet(
+                dataSetId: dataSetVersion.DataSetId,
+                indicators: ["sess_authorised"],
+                sorts: ["timePeriod|Asc", ..notFoundSorts]
+            );
+
+            var validationProblem = response.AssertValidationProblem();
+
+            Assert.Single(validationProblem.Errors);
+
+            validationProblem.AssertHasSortFieldsNotFoundError("sorts", notFoundSorts);
+        }
+
+        [Theory]
+        [InlineData("locations.in")]
+        [InlineData("locations.notIn")]
+        public async Task Locations_Empty_Returns400(string path)
+        {
+            var dataSetVersion = await SetupDefaultDataSetVersion();
+
+            var response = await QueryDataSet(
+                dataSetId: dataSetVersion.DataSetId,
+                indicators: ["sess_authorised"],
+                queryParameters: new Dictionary<string, StringValues>
+                {
+                    {
+                        $"{path}[]", ""
+                    }
+                }
+            );
+
+            var validationProblem = response.AssertValidationProblem();
+
+            Assert.Single(validationProblem.Errors);
+
+            validationProblem.AssertHasNotEmptyError(path);
+        }
+
+        [Theory]
+        [InlineData("locations.in")]
+        [InlineData("locations.notIn")]
+        public async Task Locations_InvalidMix_Returns400(string path)
+        {
+            var dataSetVersion = await SetupDefaultDataSetVersion();
+
+            var response = await QueryDataSet(
+                dataSetId: dataSetVersion.DataSetId,
+                indicators: ["sess_authorised"],
+                queryParameters: new Dictionary<string, StringValues>
+                {
+                    {
+                        path, new StringValues(
+                            [
+                                "",
+                                "invalid",
+                                "||",
+                                "LADD|code|12345",
+                                "NATT|code|12345",
+                                "NAT|invalid|12345",
+                                "LA|urn|12345",
+                                "SCH|code|12345",
```

Comment:
Might be nice to add some invalid properties for `PROV` and `RSC` too, just for completeness